cli/flox/src/commands/lock_manifest.rs

```diff
@@ -0,0 +1,59 @@
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::{Context, Result};
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::models::lockfile::Lockfile;
+use tracing::instrument;
+
+use crate::subcommand_metric;
+
+/// Lock a manifest file read from either the path apecified by `--manifest`
+/// or stdin.
+/// If provided, uses the lockfile from the path specified by `--lockfile`
+/// as the base lockfile.
+/// Returns the lockfile as JSON to stdout.
+#[derive(Bpaf, Clone)]
+pub struct LockManifest {
+    /// The manifest file to lock. (default: stdin)
+    #[bpaf(long, short, argument("path"))]
+    manifest: Option<PathBuf>,
```

Comment:
I think this should be a positional argument rather than an optional flag because it always needs a manifest file and it's surprising for `flox lock-manifest` to wait on STDIN if you call it with no arguments. It could take `-` as the positional argument to indicate STDIN, similar to `flox edit -f -`.